interfaces/builtin/audio_record.go

```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+// The audio-record interface is the companion interface to the audio-playback
+// interface and is not meant to be used without it. The design of this
+// interface is based on the idea that the slot implementation (eg pulseaudio)
+// is expected to query snapd on if the audio-record slot is connected or not
+// and the audio service will mediate recording. If other audio recording
+// servers require different security policy for record (eg, a different socket
+// path), then those accesses will be added to this interface.
+
+const audioRecordSummary = `allows operating as or interacting with recording audio services`
+
+const audioRecordBaseDeclarationSlots = `
+  audio-record:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-connection:
+      on-classic: false
+    deny-auto-connection: true
+`
+
+const audioRecordConnectedPlugAppArmor = `
+# Access for communication with audio recording service done via
+# audio-playback interface. The audio service will verify if the audio-record
+# interface is connected.
```

Comment:
The fact this interface does nothing feels so magical. Can we somehow test this via spread so we have some certainty that this is hanging together properly?

Reply:
I was thinking that we could add a new backend that serves as a indication that a trusted helper mediates access and somehow interacts with snapd to determine if the connection is present.

Some quick draft:
```
func (iface *audioPlaybackInterface) TrustedHelperConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
   spec.GrantPermission("audio-recording")
   return nil
}
```

Then we have clear indication that something is happening on our end and eventually snapd can grow an API endpoint that shows permissions for a given security label.